crates/block-producer/src/errors.rs

```diff
@@ -100,25 +100,68 @@ pub enum AddTransactionError {
     },
 }
 
+// Error codes for gRPC Status::details
+// =================================================================================================
+
+/// Error codes for `SubmitTransaction` errors
+#[repr(u8)]
+pub enum SubmitTransactionErrorCode {
+    InputNotesAlreadyConsumed = 0,
+    UnauthenticatedNotesNotFound = 1,
+    OutputNotesAlreadyExist = 2,
+    IncorrectAccountInitialCommitment = 3,
+    InvalidTransactionProof = 4,
+    StaleInputs = 5,
+    Expired = 6,
+    DeserializationFailed = 7,
+    Internal = 8,
+}
```

Comment:
I think this approach should be sufficient and that we wouldn't need json/proto encoding.

Would the rust client have access to this enum somehow? The web client would rely on some specification derived from this code?